Resync `web-animations` from WPT upstream

Ahmad-S792 opened the pull request:
#### d6dfb32962f4149af2d9755805eb8522ecc61165
<pre>
Resync `web-animations` from WPT upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=296419">https://bugs.webkit.org/show_bug.cgi?id=296419</a>
<a href="https://rdar.apple.com/problem/157172348">rdar://problem/157172348</a>

Reviewed by Antoine Quint.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/5c03a360148b834572a19c1d04592ea058f5468e">https://github.com/web-platform-tests/wpt/commit/5c03a360148b834572a19c1d04592ea058f5468e</a>

* LayoutTests/imported/w3c/web-platform-tests/web-animations/README.md:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-types.js:
(const.lengthPairType.testInterpolation):
(const.lengthPairType.testAdditionOrAccumulation):
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/background-shorthand-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/background-shorthand.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/effect-value-context-filling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/effect-value-context-filling.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/effectively-infinite-timing-parameters.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/get-keyframe-fontsize-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/non-interpolable-transition.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/sibling-index-offset-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animatable/animate-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/commitStyles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/commitStyles.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/onfinish-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/onfinish.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress-expected.txt: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/constructor-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/constructor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/DocumentTimeline/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/DocumentTimeline/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/KeyframeEffect/constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/KeyframeEffect/setKeyframes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/resources/keyframe-tests.js:
(output.keyframe):
* LayoutTests/imported/w3c/web-platform-tests/web-animations/responsive/toggle-animated-iframe-visibility.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/reverse-running-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/setting-the-start-time-of-an-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/setting-the-start-time-of-an-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/start-time-compat-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/start-time-compat.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/time-transformations/transformed-progress-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/time-transformations/transformed-progress.html:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/298947@main">https://commits.webkit.org/298947@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0899f1f86af62f933edda1c376edbfb068eb987a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117263 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36933 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27545 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123361 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69248 "Built successfully") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37629 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45520 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88987 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/43660 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120196 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29945 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105136 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69492 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23250 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67035 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99343 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23432 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126483 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44160 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33171 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97656 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44516 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101372 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97451 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24817 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42802 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20728 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44033 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43489 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46834 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45185 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/commitStyles-expected.txt:
```diff
@@ -1,29 +1,33 @@
 
-PASS Commits styles
-PASS Commits styles for individual transform properties
-PASS Commits styles for an animation that has been removed
-PASS Commits shorthand styles
-PASS Commits logical properties
-PASS Commits logical properties as physical properties
-PASS Commits values calculated mid-interval
-PASS Commits variable references as their computed values
-PASS Commits custom variables
-PASS Commits em units as pixel values
-PASS Commits relative line-height
-PASS Commits transforms
-PASS Commits transforms as a transform list
-FAIL Commits matrix-interpolated relative transforms assert_equals: Resolved transform is correct after commit. expected "matrix(2, 0, 0, 2, 100, 0)" but got "matrix(2, 0, 0, 2, 0, 0)"
-PASS Commits "none" transform
-PASS Commits the intermediate value of an animation in the middle of stack
-PASS Commit composites on top of the underlying value
-PASS Triggers mutation observers when updating style
-PASS Does NOT trigger mutation observers when the change to style is redundant
-PASS Throws if the target element is a pseudo element
```
Why did all of these get removed? Did we not export something here?

Ahmad-S792 replied:
Seems right as per WPT history and our internal test copy - 
<img width="1406" height="710" alt="image" src="https://github.com/user-attachments/assets/4912ad78-f397-42dd-85aa-5f9edc0e005f" />

annevk replied:
Okay, so they got reordered, but that doesn't really explain why we are failing more of them now. This would really benefit from a closer look by @graouts. Unless this is high priority I would suggest we wait until he's back.

graouts replied:
Hmm, it would appear that we [regressed a fair few subtests in STP 221](https://wpt.fyi/results/web-animations/interfaces/Animation/commitStyles.html?label=master&label=experimental&product=safari-220%20preview&product=safari-221%20preview). The [reordering change itself dates back to April 7](https://github.com/web-platform-tests/wpt/commit/42070d428e19bf359ba3c181e6e8d9c724dbf129) and that change [did not affect STP](https://github.com/web-platform-tests/wpt/pull/51843/checks?check_run_id=39956675967). The [next change to this test was made on June 24](https://github.com/web-platform-tests/wpt/commit/9bb63705b931d12d7f45d3f7248854ae0e115a64) right in the STP 220 → 221 interval, and the [WPF.fyi checks](https://github.com/web-platform-tests/wpt/pull/53342/checks) did not run for STP, but they did for Chrome and they show a regression in line with the failures that STP started exhibiting.

I think we should land this rebaseline, as-is and get to the bottom of why this regressed. I filed https://bugs.webkit.org/show_bug.cgi?id=297594 so that we can understand what's going on.

annevk commented on LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/setting-the-start-time-of-an-animation.html:
```diff
@@ -325,58 +325,5 @@
-  assert_start_time(animStartedDuringSecondPageRenderingUpdate,
-    timelineTimeForSecondPageRenderingUpdate,
-    "during the second page rendering update");
-}, 'Checking the start time of animations started at various times between two page rendering updates');
```
This also seems concerning, why is this all removed?

Ahmad-S792 replied:
Not exported - https://commits.webkit.org/294049@main - will do export first.

annevk replied:
Thanks!

Ahmad-S792 replied:
Exported and resynced now.

Ahmad-S792 committed:
Resync `web-animations` from WPT upstream
https://bugs.webkit.org/show_bug.cgi?id=296419
rdar://problem/157172348

Reviewed by Antoine Quint.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/5c03a360148b834572a19c1d04592ea058f5468e

* LayoutTests/imported/w3c/web-platform-tests/web-animations/README.md:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-types.js:
(const.lengthPairType.testInterpolation):
(const.lengthPairType.testAdditionOrAccumulation):
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/background-shorthand-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/background-shorthand.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/effect-value-context-filling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/effect-value-context-filling.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/keyframe-effects/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/effectively-infinite-timing-parameters.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/get-keyframe-fontsize-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/non-interpolable-transition.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/sibling-index-offset-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animatable/animate-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/commitStyles-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/commitStyles.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/onfinish-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/onfinish.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress-expected.txt: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/overallProgress.tentative.html.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/scroll-timeline-overallProgress.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/Animation/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/constructor-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/constructor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/AnimationTrigger/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/DocumentTimeline/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/DocumentTimeline/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/KeyframeEffect/constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/interfaces/KeyframeEffect/setKeyframes-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/resources/keyframe-tests.js:
(output.keyframe):
* LayoutTests/imported/w3c/web-platform-tests/web-animations/responsive/toggle-animated-iframe-visibility.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/reverse-running-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/setting-the-start-time-of-an-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/setting-the-start-time-of-an-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/start-time-compat-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/start-time-compat.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/animations/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/time-transformations/transformed-progress-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/timing-model/time-transformations/transformed-progress.html:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/298947@main
